src/java.base/unix/classes/sun/nio/fs/UnixFileAttributes.java

```diff
@@ -74,6 +74,19 @@ static UnixFileAttributes get(UnixPath path, boolean followLinks)
         return attrs;
     }
 
+    // get the UnixFileAttributes for a given file. Returns null if the file does not exist.
+    static UnixFileAttributes getIfExists(UnixPath path) throws UnixException {
+        UnixFileAttributes attrs = new UnixFileAttributes();
+        int errno = UnixNativeDispatcher.stat2(path, attrs);
```

Comment:
I think that this is the only use of `stat2()`. It could be deleted if this method were replaced with this:

```
    static UnixFileAttributes getIfExists(UnixPath path) throws UnixException {
        UnixFileAttributes attrs = new UnixFileAttributes();
        try {
            UnixNativeDispatcher.stat(path, attrs);
            return attrs;
        } catch (UnixException e) {
            if (e.errno() == UnixConstants.ENOENT) {
                return null;
            }
            throw e;
        }
    }
```

Reply:
> I think that this is the only use of `stat2()`. It could be deleted if this method were replaced with this:

The purpose of these additions to the SPI is to improve the performance of user facing methods that don't throw an exception when the file doesn't exist. It needs a stat/equivalent that doesn't throw so this is the reason for stat2. The equivalent on Windows will need to added too.

Reply:
Understood.